src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -144,23 +155,35 @@ public void finishLocalSetup() {
         childContext = checkContext;
     }
 
+    /**
+     * {@inheritDoc} Creates child module.
+     * @noinspection ChainOfInstanceofChecks
+     */
     @Override
     public void setupChild(Configuration childConf)
             throws CheckstyleException {
         final String name = childConf.getName();
         final Object module = moduleFactory.createModule(name);
-        if (!(module instanceof AbstractCheck)) {
+        if (module instanceof AutomaticBean) {
```

Comment:
lets move this nuance to #4814 .
It could be done later.

@rnveach , @MEZk , please approve PR.